gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -0,0 +1,447 @@
+      for (ZoneField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The fields of a DNS record.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@code Dns#listDnsRecords(BigInteger, DnsRecordListOption...)} or {@code
+   * Dns#listDnsRecords(String, DnsRecordListOption...)}. The name is always returned even if
+   * not selected.
+   */
+  enum DnsRecordField {
+    DNS_RECORDS("rrdatas"),
+    NAME("name"),
+    TTL("ttl"),
+    TYPE("type");
+
+    private final String selector;
+
+    DnsRecordField(String selector) {
+      this.selector = selector;
+    }
+
+    String selector() {
+      return selector;
+    }
+
+    static String selector(DnsRecordField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(NAME.selector());
+      for (DnsRecordField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The fields of a change request.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@code Dns#applyChangeRequest(ChangeRequest, BigInteger, ChangeRequestOption...)}
+   * or {@code Dns#applyChangeRequest(ChangeRequest, String, ChangeRequestOption...)}
+   * The ID is always returned even if not selected.
+   */
+  enum ChangeRequestField {
+    ID("id"),
+    START_TIME("startTime"),
+    STATUS("status"),
+    ADDITIONS("additions"),
+    DELETIONS("deletions");
+
+    private final String selector;
+
+    ChangeRequestField(String selector) {
+      this.selector = selector;
+    }
+
+    String selector() {
+      return selector;
+    }
+
+    static String selector(ChangeRequestField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(ID.selector());
+      for (ChangeRequestField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The sorting keys for listing change requests. The only currently supported sorting key is the
+   * when the change request was created.
+   */
+  enum ChangeRequestSortingKey {
+    TIME_CREATED("changeSequence");
+
+    private final String selector;
+
+    ChangeRequestSortingKey(String selector) {
+      this.selector = selector;
+    }
+
+    String selector() {
+      return selector;
+    }
+  }
+
+  /**
+   * The sorting order for listing.
+   */
+  enum SortingOrder {
+    DESCENDING, ASCENDING;
+
+    public String selector() {
+      return name().toLowerCase();
+    }
+  }
+
+  /**
+   * Class that for specifying DNS record options.
+   */
+  class DnsRecordListOption extends AbstractOption implements Serializable {
```

Comment:
Looks like most of the options are not used in a test. Maybe we should have a DnsTest that tests the options?